src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -23,6 +23,7 @@ public static class StringTests
         [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', '\0', 'i', 'j' }, "abcdefgh")]
         [InlineData(new char[] { 'a', '\0' }, "a")]
         [InlineData(new char[] { '\0' }, "")]
+        [InlineData(new char[] { '?', '@', ' ', '\0' }, "?@ ")] // ? and @ don't have overlapping bits
```

Comment:
No, I meant what's currently checked in; when I wrote that comment, I didn't think I would find another algorithm, but I did (that's when I temporarily made this no merge). Depending on whether you're OK with the other PR going forward, I'm going to add a corresponding test for it here/keep this as-is and remove the [no merge] label.